app/code/Magento/CatalogInventory/Api/StockStateInterface.php

```diff
@@ -70,4 +70,14 @@ public function suggestQty($productId, $qty, $scopeId = null);
      * @return float
      */
     public function getStockQty($productId, $scopeId = null);
+
+    /**
+     * Check whenever product qty is decimal
+     *
+     * @param string|int $productId
+     * @param string|int|null $scopeId
+     *
+     * @return bool
+     */
+    public function isStockQtyDecimal($productId, $scopeId = null): bool;
```

Comment:
We are not allowed to add new methods to the existing interfaces according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/). Consider creating a new interface with `isStockQtyDecimal ` method